### PR TITLE
feat: rework command structure

### LINKED ISSUE #202: Rewrite sync command to work with only the context created on command execution
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/bank-vaults/secret-sync/issues) for an issue that matches the one I want to file, without success.
- [X] I am not looking for support or already pursued the available [support channels](https://bank-vaults.dev/docs/support) without success.
- [X] I agree to follow the [Code of Conduct](https://bank-vaults.dev/docs/code-of-conduct).

### Version

v0.1.3

### Expected Behavior

`command.ExecuteContext(context.Background())` should create the context, and it should be retrievable with `command.Context()`.

### Actual Behavior

Currently the `NewSyncCMD` function runs `init()` and `run()` and after that the sync command is executed, therefore we cannot use `command.ExecuteContext(context.Background())` to create a context, and then use it elsewhere with `command.Context()`.

### Steps To Reproduce

_No response_

### Logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by csatib02**
## Overview

- Reworks the sync command, to follow the pattern usually used among `cobra-cli` applications.

Fixes #202

### CODE REVIEW

File: pkg/utils/logger.go
Comment (ramizpolic) [inline pkg/utils/logger.go:1]: we can move this to the `cmd` package since it's only initialised on CLI call

File: pkg/apis/v1alpha1/syncjob_types.go
Comment (ramizpolic) [inline pkg/apis/v1alpha1/syncjob_types.go:1]: we could make changes here in another PR due to PR scope that only relates to CLI changes

File: cmd/sync.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	rootCmd.AddCommand(syncCmd)
+	syncCmd.PersistentFlags().StringP(flagSource, "s", "", "Source store config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagSource)
+	syncCmd.PersistentFlags().StringP(flagTarget, "t", "", "Target store config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagTarget)
+	syncCmd.PersistentFlags().String(flagSyncJob, "", "Sync job config file. ")
+	_ = syncCmd.MarkPersistentFlagRequired(flagSyncJob)
+
+	syncCmd.PersistentFlags().String(flagSchedule, "", "Sync periodically using CRON schedule. If not specified, runs only once.")
+}
```
Comment (ramizpolic) [inline cmd/sync.go:62]: common approach is to expose command options in a form of a struct rather than using flag name consts

```suggestion
type syncJob struct {
	source *v1alpha1.StoreClient
	target *v1alpha1.StoreClient
	sync   *v1alpha1.SyncPlan
}

var syncCmd = &cobra.Command{
	Use:   "sync",
	Short: "Synchronizes secrets from a source to a target store based on sync strategy.",
	RunE:  run,
}

var syncCmdParams = struct {
	SourceStorePath string
	TargetStorePath string
	SyncJobPath     string
}{}

func init() {
	rootCmd.AddCommand(syncCmd)
	syncCmd.PersistentFlags().StringVarP(&syncCmdParams.SourceStorePath, "source", "s", "", "Source store config file. ")
	_ = syncCmd.MarkPersistentFlagRequired("source")
	// same for others
}
```

File: pkg/apis/v1alpha1/syncjob_types.go
Comment (csatib02) [reply pkg/apis/v1alpha1/syncjob_types.go:1]: I agree, that change was necessary since the SyncJob name was already in use.

File: cmd/sync.go (hunk 1 above)
Comment (csatib02) [reply cmd/sync.go:62]: I took the idea from Bank-Vaults CLI, but sure we can have a struct for params.

Comment (csatib02) [reply cmd/sync.go:62]: The idea is from the `Bank-Vaults CLI`, I believe we should avoid working with pointers on our side, and offload this to `Cobra-CLI`, by having a SSOT for the params as constants.

Comment (ramizpolic) [reply cmd/sync.go]: we could probably simplify more if we also opted in for similar approach for BV CLI

Comment (csatib02) [reply cmd/sync.go]: Added the suggested change.

**Review by ramizpolic — APPROVED** [review]
